Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/pickable_unittest.py

```diff
@@ -317,3 +322,50 @@ def test_double(self):
             '2.3@branch-b | 790725a6d79e | 7th commit\n'
             '2.2@branch-b | 3cd32e352410 | 5th commit\n',
         )
+
+    def test_related_commits_from_notes(self):
+        # Create a mock for the run method
+        mock_run = Mock()
+
+        # Simulate the output of 'git notes show'
+        mock_run.return_value = ProcessCompletion(
+            returncode=0,
+            stdout='Picked-from: abc1234def456\nCherry-picked-from: 789abc123def\nSome unrelated line\n'
+        )
+
+        commit = Mock()
+        commit.hash = 'commit_hash'
+
+        # Mock the `run` method used in the `related_commits_from_notes` method
+        related = Pickable.related_commits_from_notes(commit, run_command_function=mock_run)
```

Comment:
This probably needs to be done with the much deeper mocks in `mocks/local/git.py`. I see you started to add some. These mocks are hard to setup, but they end up being expandable. @briannafan can hopefully help, 875467dc29efa0c3 is an example of a recently added mock